api/Port.go

```diff
@@ -151,3 +175,104 @@ func (vrsConnection *VRSConnection) UpdatePortMetadata(name string, metadata map
 
 	return nil
 }
+
+// GetNuagePortTableUpdate will register with OVSDB
+// for Nuage Port table updates and return as soon as
+// port table entry gets populated
+func (vrsConnection *VRSConnection) GetPortIPv4Info() <-chan PortIPv4Info {
+	var err error
+	clientChan := make(chan PortIPv4Info, 1)
+        vrsConnection.updatesChan = make(chan *libovsdb.TableUpdates)
+        vrsConnection.ovsdbClient.Register(vrsConnection)
+
+	//set all monitors for ovsdb
+	//set a monitor on Nuage_Port_Table
+	tablesOfInterest := map[string]empty{"Nuage_Port_Table": {}}
+	monitorRequests := make(map[string]libovsdb.MonitorRequest)
+	schema, ok := vrsConnection.ovsdbClient.Schema["Open_vSwitch"]
+	if !ok {
+		return clientChan
+	}
+	for table, tableSchema := range schema.Tables {
+		if _, interesting := tablesOfInterest[table]; interesting {
+			var columns []string
+			for column := range tableSchema.Columns {
+				columns = append(columns, column)
+			}
+			monitorRequests[table] = libovsdb.MonitorRequest{
+				Columns: columns,
+				Select: libovsdb.MonitorSelect{
+					Initial: true,
+					Modify:  true}}
+		}
+	}
+	initialData, err := vrsConnection.ovsdbClient.Monitor("Open_vSwitch", nil, monitorRequests)
+	if err != nil {
+		return clientChan
+	}
+
+        portInfo := PortIPv4Info{}
+        addedTable, addedRow := getUpdatedPortTableRow(initialData, add)
+        if addedTable == true { 
+                portInfo = getPortInfo(porttable, addedRow, add)
+        }
+        if portInfo.ipaddr != "" && portInfo.gateway != "" && portInfo.mask != "" {
+        	clientChan <- portInfo
+		return clientChan
+        }
+
+	go func() {
+                for {
+                        currentUpdate := <-vrsConnection.updatesChan
+                        updatedTable, updatedRow := getUpdatedPortTableRow(currentUpdate, update)
+                        if updatedTable == true {
```

Comment:
What if updatedTable is false ? Does that imply that the update is for a port in which we are not interested ?